cms/static/js/views/pages/container.js

```diff
@@ -186,6 +186,21 @@ define(['jquery', 'underscore', 'backbone', 'gettext', 'js/views/pages/base_page
                     modal = new EditXBlockModal(options);
                 event.preventDefault();
 
+                  // WIP: now check if we want to launch in a modal for the new editors (behind waffle flag)
+                var useNewTextEditor = this.$('.edit-button').attr("use-new-editor-text"),
+                    useNewVideoEditor = this.$('.edit-button').attr("use-new-editor-video"),
+                    useNewProblemEditor = this.$('.edit-button').attr("use-new-editor-problem"),
+                    blockType = xblockElement.find('.xblock').attr("data-block-type");
```

Comment:
Have you confirmed this block type attribute shows up in the three blocks listed? Do you know what its direct purpose is? I ask just wondering if it can be trusted to always be present and provide the block type.

Reply:
yes, every block is required to have a `data-block-type.` Every block regularly configured in studio has one (thats what i checked) and "sequential" and "chapter" and "vertical" all have them as well. If it were not to find such an attribute, it would return undefined, and it would follow the old editor flow, not error, thankfully.